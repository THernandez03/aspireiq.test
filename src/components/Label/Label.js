import React from "react";
import classnames from "classnames";

import { isEmail } from "../../validations";
import { Warning, Close } from "../Icons";
import styles from "./Label.module.scss";

export const Label = ({ children, className, onIconClick = () => {} }) => {
  const isValid = isEmail(children);

  if (!children) return null;

  return (
    <div
      className={classnames(styles.wrapper, className, {
        [styles.invalid]: !isValid,
      })}
    >
      <span className={styles.text}>{children.toLowerCase()}</span>
      <div className={styles.icon} onClick={() => onIconClick()}>
        {isValid ? <Close /> : <Warning />}
      </div>
    </div>
  );
};
