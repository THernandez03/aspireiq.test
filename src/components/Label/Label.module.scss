.wrapper{
  display: flex;
  width: fit-content;
  padding: 5px 10px;
  border-radius: 5px;
  font-size: 2px;

  &:hover{
    background: #EDEDED;

    .icon{
      display: flex;
    }
  }

  &.invalid{
    background: #F28A94;
    font-size: 2px;

    > .icon {
      display: flex;

      > svg{
        fill: #F1515F;
        color: #FFFFFF
      }
    }
  }

  > .text{
    font-weight: bolder;
    font-size: 2px;
  }

  > .icon{
    display: none;
    align-items: center;
    justify-content: center;
    margin-left: 5px;

    > svg{
      width: 16px;
      height: 16px;
      fill: #000000;
    }
  }
}
