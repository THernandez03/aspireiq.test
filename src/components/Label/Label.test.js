import React from "react";
import { render } from "@testing-library/react";

import { Label } from "./Label";

describe("<Label/>", () => {
  it("should render a Label", () => {
    const { container, rerender } = render(<Label />);
    expect(container).toMatchSnapshot();
    rerender(<Label>valid@email.com</Label>);
    expect(container).toMatchSnapshot();
    rerender(<Label>invalid_email.com</Label>);
    expect(container).toMatchSnapshot();
  });
});
