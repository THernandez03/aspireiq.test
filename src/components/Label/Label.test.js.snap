// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label/> should render a Label 1`] = `<div />`;

exports[`<Label/> should render a Label 2`] = `
<div>
  <div
    class="wrapper"
  >
    <span
      class="text"
    >
      valid@email.com
    </span>
    <div
      class="icon"
    >
      <svg
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 18L18 6M6 6l12 12"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`<Label/> should render a Label 3`] = `
<div>
  <div
    class="wrapper invalid"
  >
    <span
      class="text"
    >
      invalid_email.com
    </span>
    <div
      class="icon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <path
            d="M10 6H14v12H10Z"
            fill="currentColor"
          />
          <path
            d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm-1-7v2h2v-2h-2zm0-8v6h2V7h-2z"
          />
        </g>
      </svg>
    </div>
  </div>
</div>
`;
